pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -668,12 +676,34 @@ func translateQueryParamMatch(name string, in *networking.StringMatch) route.Que
 	return out
 }
 
+// isCatchAllHeaderMatch determines if the given header is matched with all strings or not.
+// Currently, if the regex has "*" value, it returns true
+func isCatchAllHeaderMatch(in *networking.StringMatch) bool {
+	catchall := false
+
+	if in == nil {
+		return true
+	}
+
+	switch m := in.MatchType.(type) {
+	case *networking.StringMatch_Regex:
+		catchall = m.Regex == "*"
```

Comment:
> https://istio.io/latest/docs/reference/config/networking/virtual-service/#StringMatch
 RE2 style regex-based match (https://github.com/google/re2/wiki/Syntax).

This looks like incorrect, `*` not match all, but `.*` means match all ?